lib/watcher_adapter.js

```diff
@@ -27,43 +27,40 @@ class WatcherAdapter extends EventEmitter {
     if (!Array.isArray(watchedPaths)) {
       throw new TypeError(`WatcherAdapter#watch's first argument must be an array of watchedPaths`);
     }
-
     let watchers = watchedPaths.map(watchedPath => {
-      return new Promise(resolve => {
-        const watcher = new sane(watchedPath, this.options);
-        this.watchers.push(watcher);
-
-        bindFileEvent(this, watcher, 'change');
-        bindFileEvent(this, watcher, 'add');
-        bindFileEvent(this, watcher, 'delete');
-
+      const watcher = new sane(watchedPath, this.options);
+      this.watchers.push(watcher);
+      bindFileEvent(this, watcher, 'change');
+      bindFileEvent(this, watcher, 'add');
+      bindFileEvent(this, watcher, 'delete');
+      return new Promise((resolve, reject) => {
+        watcher.on('ready', resolve);
+        watcher.on('error', reject);
+      }).then(() => {
+        watcher.removeAllListeners('ready');
+        watcher.removeAllListeners('error');
         watcher.on('error', err => {
           logger.debug('error', err);
           this.emit('error', err);
         });
-
-        watcher.on('ready', () => {
-          logger.debug('ready', watchedPath);
-          resolve(watcher);
-        });
+        logger.debug('ready', watchedPath);
```

Comment:
We only want the logger.debug for the first watcher ready event?

Reply:
why? do we not want to know all the watched paths?